src/App.jsx

```diff
@@ -1,9 +1,39 @@
 import React from "react";
+//routing
+import { Routes, Route, Navigate, useLocation } from "react-router-dom";
+import { ROOT, MESSAGES, MAIN_CATEGORY } from "@routes/routeConstants.js";
+//components
+//TODO: import NAVBAR
+import Messages from "./pages/Messages.jsx";
 
 const App = () => {
   return (
     <>
-      <h1 className="text-4xl font-bold">React Boilerplate</h1>
+      <div className="h-[64px] bg-gray-500">NAVBAR COMPONENT</div>
+      {/* //TODO: when is logged in redirect to the corresponding page, else redirect to login page */}
+      <Routes>
```

Comment:
I suggest we create a routeConfig file to centralize our route configurations. This file will define all our routes, and we can then import this configuration into App.js. This approach will help us manage our routes more efficiently and keep App.js cleaner. Once you set up the routeConfig file, just link it to App.js so that all routing logic is handled there.

Reply:
I imported a route constants file, is there any example of a routeConfig file that you can show, because the boilerplate code doesn't include one?

Reply:
this is cleaner way to do it :

import React from 'react';
import { Navigate } from 'react-router-dom';
import Home from './components/Home';
import About from './components/About';
import Messages from './components/Messages';

const routeConfig = [
  {
    path: '/',
    element: <Home />,
    exact: true,
  },
  {
    path: '/about',
    element: <About />,
    exact: true,
  },
  {
    path: '/messages',
    element: <Messages />,
    exact: true,
  },
  {
    path: '*',
    element: <Navigate to="/" />,
  },
];

export default routeConfig;

import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import routeConfig from './routeConfig';

function App() {
  return (
    <Router>
      <Routes>
        {routeConfig.map((route, index) => (
          <Route key={index} path={route.path} element={route.element} exact={route.exact} />
        ))}
      </Routes>
    </Router>
  );
}

export default App;


its an example